src/jit/lower.cpp

```diff
@@ -1892,6 +1897,71 @@ void Lowering::InsertProfTailCallHook(GenTreeCall* call, GenTree* insertionPoint
     BlockRange().InsertBefore(insertionPoint, profHookNode);
 }
 
+// For fast tail calls it is necessary to set up stack args in the incoming arg space
+// on the stack. When args passed also come from this area we may run into problems
+// because we may end up overwriting the stack slot before using it. For example,
+// for foo(a, b) { return bar(b, a); }, if a and b are on incoming arg space in foo
+// they need to be swapped in this area for the call to bar. This function introduces
+// a new temp for the specified local (assumed to be in the arg space) and updates uses.
+// Note: This function may introduce new temps.
```

Comment:
Can you document the cases when a temp will be introduced?

Reply:
The text above alludes to the reason why, this sentence is mostly there to point out that this function can invalidate `LclVarDesc*`. I will flesh it out a bit more/make it more explicit (and follow the standard header format).